module/tools/RoboCupConfiguration/src/RoboCupConfiguration.cpp

```diff
@@ -0,0 +1,489 @@
+        system("systemctl restart wpa_supplicant");
+        system(("systemctl restart wpa_supplicant@" + wifi_interface).c_str());
+
+        log_message = "Network configured!";
+    }
+
+    void RoboCupConfiguration::set_config_values() {
+        /* GAME CONFIG */
+        {  // Write the robot's position to the soccer file
+            std::string soccer_file = get_config_file("Soccer.yaml");
+            // Write to the yaml file
+            YAML::Node config  = YAML::LoadFile(soccer_file);
+            config["position"] = std::string(robot_position);
+            std::ofstream file(soccer_file);
+            file << config;
+        }
+
+        {
+            // Write ready position to the corresponding config file
+            std::string ready_file   = get_config_file(robot_position.get_config_name());
+            YAML::Node config        = YAML::LoadFile(ready_file);
+            config["ready_position"] = YAML::Node(ready_position);
+            std::ofstream file(ready_file);
+            file << config;
+        }
+
+        // Check if we are on a robot
+        if (get_platform() != "nugus") {
+            log_message = "Configure Error: Network configuration only available on NUgus robots!";
+            return;
+        }
+
+        // Check if any of ip_address, ssid or password are empty
+        if (ip_address.empty() || ssid.empty() || password.empty()) {
+            log_message = "Configure Error: IP Address, SSID and Password must be set!";
+            return;
+        }
+
+        /* WIRELESS NETWORK SYSTEMD CONFIG */
+
+        // Get folder name
+        const std::string systemd_folder = fmt::format("system/{}/etc/systemd/network", hostname);
+
+        // Parse the IP address
+        std::stringstream ss(ip_address);
+        std::vector<std::string> ip_parts{};
+        for (std::string part; std::getline(ss, part, '.');) {
+            ip_parts.push_back(part);
+        }
+
+        // Check if team_id and player_id match with third and fourth parts of the IP address.
+        // This shouldn't break things, but the user should know that there may be an issue.
+        // In the lab, the IP addresses of the robots are set and may not match in this way,
+        // but at RoboCup, it is important that this format is followed
+        if (team_id != std::stoi(ip_parts[2]) || player_id != std::stoi(ip_parts[3])) {
+            log_message +=
+                "Warning: At RoboCup, the third position of the IP address should be the team ID and the fourth should "
+                "be the player number. ";
+        }
+
+        // Write the new ip address to the file
+        std::ofstream(fmt::format("{}/30-wifi.network", systemd_folder))
+            << fmt::format("[Match]\nName={}\n\n[Network]\nAddress={}/16\nGateway={}.{}.3.1\nDNS=8.8.8.8",
+                           wifi_interface,
+                           ip_address,
+                           ip_parts[0],
+                           ip_parts[1]);
```

Comment:
I copied the existing files so idk if there's things here that need to be dynamically changed too.